src/librustc_resolve/lib.rs

```diff
@@ -4865,6 +4874,7 @@ impl<'a> Resolver<'a> {
             (TypeNS, _) if old_binding.is_extern_crate() => "extern crate",
             (TypeNS, Some(module)) if module.is_normal() => "module",
             (TypeNS, Some(module)) if module.is_trait() => "trait",
+            (TypeNS, Some(module)) if module.is_trait_alias() => "trait alias",
```

Comment:
There's a possible other pattern to add in https://github.com/rust-lang/rust/pull/59166/files#diff-aeb0880081a991f34aef2ab889e1fb7aR4848, but I can't for the life of me figure out how you could be resolve imports inside a trait.

Reply:
Inside a trait alias you mean? Yeah, I can't see how either. Still, just in *case* (you don't want it falling back to "enum" for a trait alias, even if it's just a bug), let's add this case.

Reply:
No no, I don't even know the regular `trait` case could happen. There's no way to `use Something` inside `trait Wee { .. }` (except inside a function, which I assume is the `Block` case).

Reply:
Yeah, I agree... it doesn't seem possible. Maybe this is just future-proofing it? Add it in, since it can't hurt, I say!

Reply:
`report_conflict` is used for conflicts not involving imports as well, e.g.
```rust
trait Tr {
    fn foo();
    fn foo(); // ERROR the name `foo` is defined multiple times
}
```
, so `module.is_trait()` can occur.

Reply:
`module.is_trait_alias()` on the other hand cannot occur because trait aliases are not `Module`s in resolve sense, unlike traits.
(Only `Mod`s, `Enum`s and `Trait`s are `Module`s.)

Reply:
The whole `match` should be replaceable with `old_binding.descr()`, I think.